src/realm/alloc_slab.cpp

```diff
@@ -582,8 +582,10 @@ ref_type SlabAlloc::get_top_ref(const char* buffer, size_t len)
 
 namespace {
 
-std::map<std::string, std::weak_ptr<SlabAlloc::MappedFile>> all_files;
-util::Mutex all_files_mutex;
+// prevent destruction at exit (which can lead to races if other threads are still running)
+std::map<std::string, std::weak_ptr<SlabAlloc::MappedFile>>& all_files =
```

Comment:
These declarations could be good places to use `auto` as it would avoid repeating the type name.